piControlMain.c

```diff
@@ -1447,11 +1447,14 @@ static long piControlIoctl(struct file *file, unsigned int prg_nr, unsigned long
 				return -EPERM;
 			}
 
+			rt_mutex_lock(&piDev_g.lockIoctl);
 			if (!isRunning()) {
 				printUserMsg(priv, "piControl is not running");
```

Comment:
as the rt_mutex_lock is defined "__rt_mutex_lock_common(lock, TASK_UNINTERRUPTIBLE, NULL, 0)", is it totally problem free?

Reply:
Thats no problem, printk can be called from any context